runtime/src/accounts_db.rs

```diff
@@ -921,6 +949,96 @@ impl AccountsDB {
         reclaims
     }
 
+    fn squash_uncleaned_keys(&self, max_slot: Slot) {
+        let (keys, mut slots) = self.collect_uncleaned_keys_to_slot(max_slot);
+        if slots.is_empty() {
+            return;
+        }
+
+        slots.sort_unstable();
+        let max_slot = *slots.last().unwrap();
+        let mut squashed_keys = self.squashed_uncleaned_keys.write().unwrap();
```

Comment:
how about destructuring?

```rust
let (mut pubkeys, root) =  self.squashed_uncleaned_keys.write().unwrap();
```

Reply:
I didn't get it to work. The `write().unwrap()` returns RwLockGuard which I don't think can be directly destructured like that. Maybe there is another syntax, but you're kind of operating on the reference to the lock guard, so you can't copy items out to work on them.